packages/solid/src/index.ts

```diff
@@ -30,7 +30,7 @@ export type {
   Accessor,
   Setter,
   Resource,
-  ResourceReturn,
+  ResourceState,
```

Comment:
This is the only breaking change (not covered by the tests). The `ResourceReturn` and `Resource` types were not being used in the source code, but only exported. I'm not sure if anyone was relying on these unused-yet-exported types.

I renamed them to align with the API, f.e. `createResource()` now returns the type `Resource`, which happens to be a tuple.

Is this ok? I didn't like the name of `ResourceReturn`.

I did similar with renaming the internal (this time not exported) `Signal` type to `SignalState`, so that the `createSignal` function could return the new `Signal` type (a tuple) while `SignalState` is used only internally.

I think the names make more sense like this.

Another issue, that I didn't change, is that the word "accessor" in JavaScript tends to mean a getter/setter pair, and the name of the current `Accessor` type does not align with this. I think `Getter` would be a better name for it, but this would probably definition be a breaking change.

Does Solid intend to freeze at 1.0 for a long time? Or will the plan be to move it along major version bumps and keep it fresh and new?

I'm a fan of the latter, and one such library example is [three.js](https://threejs.org/) which keeps progressing for the better and not allowing itself to become stale. The three.js core team makes sure to write a migration guide in every release note, and they often break things by first deprecating old APIs for a while while the new APIs are put in place, then eventually removing them. Another method they have is that they might disable a feature, leave it behind a flag for a while, then remove it, and people who migrate it need to only set the flag to true so that they can keep using the feature for a while but eventually they know they will have to stop using it if they wish to keep upgrading.

Based on this, what we might be able to do is add the `@depracted` JSDoc comment to `Accessor`, and add a new `Getter` type. Then at some point (like after a year?) remove it.

Reply:
Yes, the unused Resource types were meant to be used in solid-primitives.